### DISCUSSION TITLE
Failure in Analysis using PyAEDT

### DISCUSSION THREAD

**Discussion by ghost** [ansys/pyaedt#3817] [Q&A] [answered]
Hi,

I'm getting errors when analysis in HFSS, so I would like to resolve them.

Using PyAEDT, I'm trying to analyze after changing the value of variable 'L1'.
The code is following:

`import sys`
`import atexit`
`from pyaedt import *`
`import pyaedt`
`pyaedt.settings.use_grpc_api=False`

`version = '2022.2'`
`print("Loading the PyAEDT Console.")`

`hfss = Hfss(projectname="antenna.aedt", designname="Design1", specified_version=version, new_desktop_session=False, non_graphical=False, close_on_exit=False, student_version=False)`

`hfss["L1"] = "26.5mm"`

`hfss.analyze_setup("Setup1")`

It works fine up to the point where the value of variable 'L1' is changed.
Setup1 is already set up when I designed the antenna model.

It displays the following error message when I carry out the program.

> PyAEDT INFO: Key Desktop/ActiveDSOConfigurations/HFSS correctly changed.
> PyAEDT INFO: Solving design setup Setup1
> PyAEDT INFO: Key Desktop/ActiveDSOConfigurations/HFSS correctly changed.
> PyAEDT ERROR: Error in Solving Setup Setup1
> False

The result was same when I entered `hfss.analyze()` instead of `hfss.analyze_setup("Setup1"),` by the way.

Can anyone tell me what is wrong with this code?

Thank you.

**Comment by Samuelopez-ansys** [comment]
Hi @crapemyrtle-841 ,

It looks like something in the design is incorrect when you change this variable, did you validate the design before analyzing it?

Please share the project in a zip file and the community will be able to help you. If you have problems with the validation, it is not a PyAEDT problem, please contact Ansys customer support in that case.

**Reply by ghost** [reply]
Hi @Samuelopez-ansys 

Thank you for your response.
I attached a zip file.
[antenna.aedt.zip](https://github.com/ansys/pyaedt/files/13226538/antenna.aedt.zip)

It worked correctly when I analyzed this model in HFSS without PyAEDT.
But, it displays the error when using PyAEDT.
Please tell me why it doesn't work correctly.

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @crapemyrtle-841 ,

It is working fine for me:

<img width="1838" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/fd55c1fa-c19a-435c-aa23-697fb718a571">


```
from pyaedt import *
import pyaedt

pyaedt.settings.use_grpc_api = False

version = '2023.2'
print("Loading the PyAEDT Console.")
project = r"C:\Users\slopez\Downloads\antenna_test.aedt"
hfss = Hfss(projectname=project, designname="Design1", specified_version=version, new_desktop_session=False,
            non_graphical=False, close_on_exit=False, student_version=False)

hfss["L1"] = "26.5mm"

hfss.analyze_setup("Setup1", num_cores=4)

hfss.release_desktop(False, False)
```

Please update pyaedt, because probably you are using an old version, I am using pyaedt v0.7.3

**Reply by ghost** [reply]
Hi @Samuelopez-ansys

Thank you for running it.
It worked fine by using `num_cores=4`.
Thank you so much!
